core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerConfig.java

```diff
@@ -126,14 +121,6 @@ public class RemoteStorageManagerConfig extends AbstractConfig {
             CHUNK_SIZE_DOC
         );
 
-        CONFIG.define(
-            CHUNK_CACHE_CONFIG,
-            ConfigDef.Type.CLASS,
-            null,
-            ConfigDef.Importance.MEDIUM,
-            CHUNK_CACHE_DOC
-        );
-
```

Comment:
The reasoning is rather simple, it's not really possible to instatiate Caffeine without exposing most of the implementation details of different implementations to the config since configuration depends on method implemented in specific subclasses. And factory brings additional convinies by incapsulating implementation details of ChunkManagers into separate place and not polluting configs/RSM with it.

Reply:
I could be possible to make it configurable with classes if we refactor this a bit, though. For example, create `AbstractChunkManager` that would every other inherit from (maybe indirectly) and move `getChunk` logic there. Thus, we can get rid of wrapping and all the non-abstract classes will have the uniform constructor with `final ObjectFetcher fetcher, final ObjectKey objectKey, final AesEncryptionProvider aesEncryptionProvider` args.